### ISSUE TITLE
feature request - print to console with office script

### ISSUE THREAD

**Issue by Jeroendevr** [xlwings/xlwings#2231]
#### Describe your issue (incl. Traceback!)
Within the office scripts domain, I would like to print to the console some (custom) exceptions. This to aid end users if there is missing data, a sheet is missing etc. 

Currently the developer is only the one who can debug an issue. 

Example of printing the current sheet to the console.
```js
function main(workbook: ExcelScript.Workbook) {
    // Get the active worksheet.
    let sheet = workbook.getActiveWorksheet();

    // Display the current worksheet's name.
    console.log(sheet.getName());
}
```

Running a runPython function often yield

```sh
Line 221: Load failed
```
Which points to 

```js
    // API call
    let response = await fetch(url, {
        method: "POST",
        headers: headers,
        body: JSON.stringify(payload),
    });
```

**Comment by fzumstein** [comment]
I need to look into `Load failed` as I believe this worked correctly once, so either a regression or they changed something in the JS runtime. Anyhow, for now, Office Scripts is super limited, so the only way you can show something to the end user is either by printing it on a sheet (you could insert and activate a new sheet) or by using `book.app.alert()`, which currently throws a JS error, so not totally ideal, but might do for certain use cases. console.log is invisible to the user if they just run the script without being in the editor.

**Comment by fzumstein** [comment]
Ok I remember now: the issue is caused by CORS as FastAPI/Starlette doesn't slap the headers on responses caught via exception handler: https://github.com/encode/starlette/issues/1175

There's a solution suggested in the issue, or adding the headers manually in the exception handler would also work:

```python
@app.exception_handler(Exception)   # this handles every exception
async def exception_handler(request, exception):
    headers = {"Access-Control-Allow-Origin": "*"}
    return PlainTextResponse(
        str(exception), status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
        headers=headers,
    )
```

**Comment by Jeroendevr** [comment]
Great it did work. For future users don't forget to import relevant starlette. 

```python
from starlette.responses import PlainTextResponse
from starlette import status

@app.exception_handler(Exception)   # this handles every exception
async def exception_handler(request, exception):
    headers = {"Access-Control-Allow-Origin": "*"}
    return PlainTextResponse(
        str(exception), status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
        headers=headers,
    )
```

Is this something that is going to be fixed by starlette or would it better to note this code in the docs?

**Comment by fzumstein** [comment]
Ok, so Starlette, the web framework that FastAPI uses under the hood, doesn't add the CORS headers to unhandled Exceptions by design. The same is true if you handle the root `Exception` (other frameworks like Flask and Django don't show this behavior). This means that you'll either:

* Have to explicitly handle all Errors that you want to show in Office Scripts via an exception_handler or
* You can wrap the whole FastAPI app with the CORSMiddleware instead of using the add_middleware functionality.

The latter option is what I have chosen for the various quickstart projects, such as:

* CLI: https://github.com/xlwings/xlwings/pull/2236
* helloworld: https://github.com/xlwings/xlwings-server-helloworld-fastapi/blob/main/app/main.py
* officejs quickstart: https://github.com/xlwings/xlwings-officejs-quickstart/blob/main/app/server_fastapi.py

In summary, ignore my previous comment and don't add the Access-Control-Allow-Origin manually. Also, I'll add at some point a "production check list" in the docs where a more strict configuration of exception handlers and CORS settings can be described, but for now I believe we can close this issue.
